runtime/compiler/optimizer/J9ValuePropagation.cpp

```diff
@@ -2188,20 +2188,33 @@ J9::ValuePropagation::innerConstrainAcall(TR::Node *node)
          static char *enableDynamicObjectClone = feGetEnv("TR_enableDynamicObjectClone");
          // Dynamic cloning kicks in when we attempt to make direct call to Object.clone
          // or J9VMInternals.primitiveClone where the cloned object is an array.
+         bool vectorSupportIntrinsic =
+            method->getRecognizedMethod() == TR::jdk_internal_vm_vector_VectorSupport_load ||
+            method->getRecognizedMethod() == TR::jdk_internal_vm_vector_VectorSupport_binaryOp ||
+            method->getRecognizedMethod() == TR::jdk_internal_vm_vector_VectorSupport_store;
+         
          if (method->getRecognizedMethod() == TR::java_lang_Object_clone
-             || method->getRecognizedMethod() == TR::java_lang_J9VMInternals_primitiveClone)
+             || method->getRecognizedMethod() == TR::java_lang_J9VMInternals_primitiveClone
+             || vectorSupportIntrinsic)
             {
             bool isGlobal;
-            if (method->getRecognizedMethod() == TR::java_lang_Object_clone)
+            if (method->getRecognizedMethod() == TR::jdk_internal_vm_vector_VectorSupport_binaryOp)
```

Comment:
Can there be a comment added here for explaining the different shape of the trees and why we are looking at different children in these different cases ?

Reply:
Thanks. Added some explanation and also removed store() intrinsic since it does not return an object. 
I have to say that, in contrast to the clone() method, the parameter whose type we need to propagate to the return value is not an instance of an object but a class. For example, the first parameter of load() intrinsic will be `Float128Vector.class` and not an instance of `Float128Vector`. But somehow it seems to be working the way we want to.